### DISCUSSION TITLE
trivy does not recognize vulnerabilities in jdk images (zulu/openjdk)

### DISCUSSION THREAD

**Discussion by nobletrout** [aquasecurity/trivy#4238] [Ideas]
according to oracle's risk matrix:
https://www.oracle.com/security-alerts/cpujan2023.html#AppendixJAVA
CVE-2023-21835 should be present in java 11.0.17

```
$trivy image -q azul/zulu-openjdk:11.0.17 | grep 2023-21835
$
```
this from the image sourced from here:
https://hub.docker.com/layers/azul/zulu-openjdk/11.0.17/images/sha256-c9aa6d80566530a395b31cc3bb2fb54eba4584438e38dc5adb4ad10eb4976dbb?context=explore

i've also tested openjdk and not found java vulnerabilities present there either.

**Comment by knqyf263** [comment]
Trivy currently doesn't support unpackaged Java, which was not installed by package managers like `dnf` and `apt`.

**Reply by breun** [reply]
Java is not unpackaged in `azul/zulu-openjdk:11.0.17`, but installed as APT packages from [Azul's APT repository](https://repos.azulsystems.com):

```
❯ docker run -it --rm azul/zulu-openjdk:11.0.17
root@dca32592092e:/# dpkg -l | grep zulu
ii  zulu-repo                  1.0.0-3                      all          Azul Systems Zulu JDK new repository setup
ii  zulu11-ca-doc              11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) CA JDK documentation
ii  zulu11-ca-jdk              11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) CA JDK
ii  zulu11-ca-jdk-headless     11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) CA Headless JDK
ii  zulu11-ca-jre              11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) CA JRE
ii  zulu11-ca-jre-headless     11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) CA Headless JRE
ii  zulu11-doc                 11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) JDK documentation
ii  zulu11-jdk                 11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) JDK
ii  zulu11-jdk-headless        11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) Headless JDK
ii  zulu11-jre                 11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) JRE
ii  zulu11-jre-headless        11.0.17-1                    amd64        Azul Zulu 11.60+19 (11.0.17-b8) Headless JRE
```

**Comment by nobletrout** [comment]
Should I change this issue type to feature request and update title or make a new one?

**Reply by knqyf263** [reply]
I changed it to the discussion because we have no idea how to achieve that. We need to identify the detail, such as OpenJDK vs Oracle JDK and the version.

**Comment by nobletrout** [comment]
FWIW I just normally run `java -version` I know that's not the most scientific solution, but it works for me

**Reply by knqyf263** [reply]
Trivy statically analyzers container images regardless of the host platform. Java might be created for a different architecture. e.g. run Trivy on arm64 against an amd64 image
